ruby/tests/basic.rb

```diff
@@ -383,10 +383,30 @@ def test_rptfield_array_ducktyping
       length_methods.each do |lm|
         assert l.send(lm)  == 0
       end
+      # out of bounds returns a nil
+      assert l[0] == nil
+      assert l[1] == nil
+      assert l[-1] == nil
+
       l.push 4
       length_methods.each do |lm|
-        assert l.send(lm)  == 1
+        assert l.send(lm) == 1
       end
+      assert l[0] == 4
+      assert l[1] == nil
+      assert l[-1] == 4
+      assert l[-2] == nil
+
+      l.push 2
+      length_methods.each do |lm|
+        assert l.send(lm) == 2
+      end
+      assert l[0] == 4
+      assert l[1] == 2
+      assert l[2] == nil
+      assert l[-1] == 2
+      assert l[-2] == 4
+      assert l[-3] == nil
     end
 
     def test_map_basic
```

Comment:
For now I think it's fine (best to do one thing at a time in a PR), but feel free to send a separate cleanup PR that splits `basic.rb` into categories of tests if you feel strongly about it. Probably would be best to split out the test message type definitions into another file included by these separate test files, if you do choose to do this.